cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -730,6 +730,8 @@ struct set_final_row_count {
     size_t const page_start_row = chunk.start_row + page.chunk_row;
     size_t const chunk_last_row = chunk.start_row + chunk.num_rows;
     page.num_rows               = chunk_last_row - page_start_row;
+    // Mark `is_num_rows_adjusted` to signal string decoders to treat this page as a bounds page.
+    page.is_num_rows_adjusted = true;
```

Comment:
Should this only be set true if the new count differs from the old?
```suggestion
    auto const adj_rows         = chunk_last_row - page_start_row;
    if (adj_rows != page.num_rows) {
      // Mark `is_num_rows_adjusted` to signal string decoders to treat this page as a bounds page.
      page.is_num_rows_adjusted = true;
      page.num_rows             = adj_rows;
    }
```

Reply:
All done!